testutil/test_helper.go

```diff
@@ -93,6 +95,31 @@ func NewTestHelper() *TestHelper {
 	return testHelper
 }
 
+// RetryFor retries a given function every second until the function returns
+// without an error, or a timeout is reached. If the timeout is reached, it
+// returns the last error received from the function.
+func RetryFor(timeout time.Duration, fn func() error) error {
```

Comment:
👍 for removing this from `*TestHelper`

Reply:
My preference is to revert this change, and leave `RetryFor` as a function on the `TestHelper` struct. One of the main motivations behind having the `TestHelper` struct is to prevent the proliferation of a bunch of global helper functions.

To add retryability to the `KubernetesHelper` struct, I recommend passing the `RetryFor` func as a parameter when initializing that struct. That way it's available to both structs but only defined in one place. Something like this:

```diff
diff --git a/testutil/kubernetes_helper.go b/testutil/kubernetes_helper.go
index 55fe82ed..ffcbf68c 100644
--- a/testutil/kubernetes_helper.go
+++ b/testutil/kubernetes_helper.go
@@ -5,6 +5,7 @@ import (
 	"os/exec"
 	"regexp"
 	"strings"
+	"time"
 
 	"github.com/linkerd/linkerd2/pkg/k8s"
 	coreV1 "k8s.io/api/core/v1"
@@ -20,10 +21,11 @@ import (
 // Kubernetes API using the environment's configured kubeconfig file.
 type KubernetesHelper struct {
 	clientset *kubernetes.Clientset
+	RetryFor  func(time.Duration, func() error) error
 }
 
 // NewKubernetesHelper creates a new instance of KubernetesHelper.
-func NewKubernetesHelper() (*KubernetesHelper, error) {
+func NewKubernetesHelper(retryFor func(time.Duration, func() error) error) (*KubernetesHelper, error) {
 	rules := clientcmd.NewDefaultClientConfigLoadingRules()
 	overrides := &clientcmd.ConfigOverrides{}
 	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
@@ -39,6 +41,7 @@ func NewKubernetesHelper() (*KubernetesHelper, error) {
 
 	return &KubernetesHelper{
 		clientset: clientset,
+		RetryFor:  retryFor,
 	}, nil
 }
 
diff --git a/testutil/test_helper.go b/testutil/test_helper.go
index 47aa711f..4f56165a 100644
--- a/testutil/test_helper.go
+++ b/testutil/test_helper.go
@@ -80,7 +80,7 @@ func NewTestHelper() *TestHelper {
 	}
 	testHelper.version = strings.TrimSpace(version)
 
-	kubernetesHelper, err := NewKubernetesHelper()
+	kubernetesHelper, err := NewKubernetesHelper(testHelper.RetryFor)
 	if err != nil {
 		exit(1, "error creating kubernetes helper: "+err.Error())
 	}
```

(For a bit of additional background: we originally had only one `*Helper` struct, but decided to split it into two based on the types of helpers they were providing, as a way to keep things organized. I'm not actually sure we benefit all that much from having two separate structs, however, so in the long run we may end up collapsing them back into a single struct, if it turns out there's a lot of overlap among the helpers. But in the near term passing `RetryFor` as a param seems like an ok solution to me.)